src/decoder/plugins/FluidsynthDecoderPlugin.cxx

```diff
@@ -109,6 +122,9 @@ fluidsynth_file_decode(DecoderClient &client, Path path_fs)
 		return;
 
 	fluid_settings_setnum(settings, setting_sample_rate, sample_rate);
+	if (gain != 0.0) {
```

Comment:
This will trigger `-Wfloat-equal`, because such a comparison is fragile. Can you use a "magic" value that's robust to compare, such as a negative value or NaN?

Reply:
@MaxKellermann What do you think of just doing `gain_set` I pushed a PR for it.

I first tried to do NaN, but I was worried about portability issues. The right thing to do is query `is_iec559` before using NaN, so I don't think I could depend on that.

Then magic values as well have some corner cases. If you would prefer I use a magic value happy to change it (what value should I do?)

Reply:
`gain_set` is fine, it's just slightly more complicated than using `std::optional`.